packages/utils/src/validateProperties.ts

```diff
@@ -0,0 +1,37 @@
+import { logger } from './logger';
+const MAX_PROPERTY_KEYS = 1000;
+
+const _isValidObject = (properties: { [key: string]: any }): boolean => {
+  if (Object.keys(properties).length > MAX_PROPERTY_KEYS) {
+    logger.warn('too many properties. Skipping operation');
+    return false;
+  }
+  for (const key in properties) {
+    if (typeof key !== 'string') {
+      logger.warn('invalid properties format. Skipping operation');
+      return false;
+    }
+    let value = properties[key];
+    if (!isValidProperties(key, value)) return false;
+  }
+  return true;
+};
+
+const isValidProperties = (property: string, value: any): boolean => {
+  if (typeof property != 'string') return false;
+  if (Array.isArray(value)) {
+    for (const valueElement of value) {
+      if (!(typeof valueElement === 'number' || typeof valueElement === 'string')) {
+        logger.warn('invalid array element type ', typeof valueElement);
```

Comment:
the identify API does allow for objects nested within arrays - just not arrays nested in arrays